### ISSUE TITLE
Error when attempting to upload a file that exceeds 'maxBytes' limit

### ISSUE THREAD

**Issue by koukuko** [balderdashy/skipper#36]
Error when attempting to upload a file that exceeds 'maxBytes' limit, is there anything wrong with skipper or is it my problem? 

```
E:\h.acfun.tv.admin\node_modules\sails\node_modules\skipper\node_modules\skipper-disk\standalone\build-progress-stream.js:129
 Garbage collecting file `' + __newFile.filename + '` located @ ' + fd + '...'
                                                                    ^
ReferenceError: fd is not defined
    at gc (E:\h.acfun.tv.admin\node_modules\sails\node_modules\skipper\node_modules\skipper-disk\standalone\build-progress-stream.js:129:96)
    at Transform.singleFileProgress (E:\h.acfun.tv.admin\node_modules\sails\node_modules\skipper\node_modules\skipper-disk\standalone\build-progress-stream.js:134:9)
    at Transform.emit (events.js:95:17)
    at Transform.__progress__._transform (E:\h.acfun.tv.admin\node_modules\sails\node_modules\skipper\node_modules\skipper-disk\standalone\build-progress-stream.js:42:10)
    at Transform._read (_stream_transform.js:179:10)
    at Transform.Readable.read (_stream_readable.js:323:10)
    at flow (_stream_readable.js:591:52)
    at WriteStream.<anonymous> (_stream_readable.js:575:7)
    at WriteStream.emit (events.js:92:17)
    at onwriteDrain (_stream_writable.js:289:12)
    at afterWrite (_stream_writable.js:277:5)
```

**Comment by ameilland** [comment]
Hi,

I still have the error.

I have sails js with skipper 0.5.5 and skipper-disk 0.5.4.

I  try to upload a csv of 45 MB.

The error is raised:

 var outs__ = fsx.createWriteStream(__newFile.fd, encoding);

            // When the file is done writing, call the callback
            outs__.on('finish', function successfullyWroteFile() {
                log('finished file: ' + __newFile.filename);
                done();
            });
            outs__.on('E_EXCEEDS_UPLOAD_LIMIT', function (err) {
                done(err);
            });

The detail is:

code: "E_EXCEEDS_UPLOAD_LIMIT"
maxBytes: 15000000
message: "Upload limit of 15000000 bytes exceeded (15048979 bytes written)"
name: "Upload Error"
stack: undefined
written: 15048979

An idea on the fix?

Thanks,

**Comment by sailsbot** [comment]
Thanks for posting, @koukuko.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by RWOverdijk** [comment]
Bump. How to fix this?

**Comment by sailsbot** [comment]
Thanks for posting, @koukuko.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by wasbridge** [comment]
I have the same problem as @ameilland how can I make it so that skipper gives me the error so that I can inform the user?

**Comment by trungducng** [comment]
Still error till now
Any fix please?

I'm working on sails ver 0.12

**Comment by cubetechnology** [comment]
@trungducng just trying to fix this i put  in ur upload options maxBytes:10000000000000, and works. i just put 10000000000000 i think if u need < than that should work. let me know if u solved with this
